### ISSUE TITLE
Success Criterion 4.1.3: Status Messages (Level AA)

### ISSUE THREAD

**Issue by Lboniello** [w3c/wcag2ict#122]
***
 

From [Success Criterion 4.1.3](https://www.w3.org/WAI/standards-guidelines/wcag/new-in-21/#413-status-messages-aa) for the SC in WCAG 2.2):
 
>In content implemented using markup languages, [status messages](https://www.w3.org/TR/WCAG22/#dfn-status-messages) can be [programmatically determined](https://www.w3.org/TR/WCAG22/#dfn-programmatically-determinable) through [role](https://www.w3.org/TR/WCAG22/#dfn-role) or properties such that they can be presented to the user by [assistive technologies](https://www.w3.org/TR/WCAG22/#dfn-assistive-technologies) without receiving focus.

##### Guidance When Applying Success Criterion 4.1.3 to Non-Web Documents and Software
   - This applies directly as written, and as described in [Intent from Understanding Success Criterion 4.1.3](https://www.w3.org/WAI/WCAG21/Understanding/status-messages.html#intent) (also provided below).
   
 ***NOTE
The user need exists to present status messages to the user programmatically, primarily through markup language, though other means may be required due to the availability of accessibility services.

***NOTE
See also the discussion on [Closed Functionality](https://w3c.github.io/wcag2ict/#closed-functionality). 
--------------------------------------------------------------------------------

In Appendix A. Success Criteria Problematic for Closed Functionality add the following:

- 4.1.3 Status Messages identifies programmatic identification. 
    -4.1.3 is problematic because it talks about assistive technology, programmatic information, and mark up language which are not always applicable to items with closed functionality.

**Note:** Non web technology with closed functionality would need equivalent facilitation in terms of providing immediate focus or access to the status messages.

***

**Comment by mraccess77** [comment]
The idea with many status messages is that it's meant in situations where you may not want to move focus - in particular moving focus without user initiation is problematic.  We should carefully craft any language around that to not imply conflict with other criterion such as SC 3.2.2.

**Comment by mraccess77** [comment]
I wonder if this still leaves confusion around what uses markup languages and if the status message has to be implemented in the markup language in order to be applicable here.  As discussed before, I believe addition of markup languages was meant to side step documents including PDFs but inadvertently removed this from being applicable to much non-closed software.

**Comment by maryjom** [comment]
@mraccess77 You caught me mid-editing of my last comment where I made some proposed changes to the first note. Let's see if this addresses your latest comment.

**Comment by mraccess77** [comment]
@maryjom thanks, yes, thank you!

**Comment by mitchellevan** [comment]
In the first note, "language" should be "languages", so the note will say:

NOTE
For non-web documents and software that are not implemented using markup <ins>languages</ins>, there is still a user need to have status messages be programmatically exposed so that they can be presented to the user by assistive technologies without receiving focus. This is typically enabled through the use of accessibility services of the user agent or platform software.

In Appendix A. Success Criteria Problematic for Closed Functionality, "mark up language" should be "markup languages", so the bullet will say:

* [4.1.3 Status Messages](https://www.w3.org/TR/WCAG22/#status-messages) — requires information in a programmatic identifiable form. Additionally, software with closed functionality is not typically implemented using <ins>markup languages</ins>.

**Comment by maryjom** [comment]
Incorporating the Task Force's approved changes from the 30 March meeting. Note, definition links won't work in this issue, as they are relative to the document.

From [Success Criterion 4.1.3](https://www.w3.org/TR/WCAG22/#status-messages):
 
>In content implemented using markup languages, [status messages](https://www.w3.org/TR/WCAG22/#dfn-status-messages) can be [programmatically determined](https://www.w3.org/TR/WCAG22/#dfn-programmatically-determinable) through [role](https://www.w3.org/TR/WCAG22/#dfn-role) or properties such that they can be presented to the user by [assistive technologies](https://www.w3.org/TR/WCAG22/#dfn-assistive-technologies) without receiving focus.

##### Guidance When Applying Success Criterion 4.1.3 to Non-Web Documents and Software
This applies directly as written, and as described in [Intent from Understanding Success Criterion 4.1.3](https://www.w3.org/WAI/WCAG21/Understanding/status-messages.html#intent) (also provided below) replacing "In content implemented using markup languages" with "In content implemented using markup languages, or that supports status message notifications".

With these substitutions, it would read:
In **[content implemented using markup languages, or that supports status message notifications]**, [status messages](https://www.w3.org/TR/WCAG22/#dfn-status-messages) can be [programmatically determined](#dfn-programmatically-determinable) through [role](#dfn-role) or properties such that they can be presented to the user by [assistive technologies](#dfn-assistive-technologies) without receiving focus.
   
**NOTE**
For non-web documents and software that are not implemented using markup languages, there is still a user need to have status messages be programmatically exposed so that they can be presented to the user by assistive technologies without receiving focus. This is typically enabled through the use of accessibility services of the user agent or platform software.

**NOTE**
See also the discussion on [Closed Functionality](https://w3c.github.io/wcag2ict/#closed-functionality). 

***
In Appendix A. Success Criteria Problematic for Closed Functionality add the following:

- [4.1.3 Status Messages](https://www.w3.org/TR/WCAG22/#status-messages) — requires information in a programmatic identifiable form. Additionally, software with closed functionality is not typically implemented using markup languages.

**NOTE** Non-web software with closed functionality would need equivalent facilitation to provide access to status messages.
